sklearn/utils/estimator_checks.py

```diff
@@ -46,6 +48,140 @@
+    # test classfiers can handle non-array data
+    yield check_classifier_data_not_an_array
+    # test classifiers trained on a single label always return this label
+    yield check_classifiers_one_label
+    yield check_classifiers_classes
+    yield check_classifiers_pickle
+    yield check_estimators_partial_fit_n_features
+    # basic consistency testing
+    yield check_classifiers_train
+    if (name not in ["MultinomialNB", "LabelPropagation", "LabelSpreading"]
+        # TODO some complication with -1 label
+            and name not in ["DecisionTreeClassifier",
+                             "ExtraTreeClassifier"]):
+            # We don't raise a warning in these classifiers, as
+            # the column y interface is used by the forests.
+
+        # test if classifiers can cope with y.shape = (n_samples, 1)
+        yield check_classifiers_input_shapes
+    # test if NotFittedError is raised
+    yield check_estimators_unfitted
+    if 'class_weight' in Classifier().get_params().keys():
+        yield check_class_weight_classifiers
+
+
+def _yield_regressor_checks(name, Regressor):
+    # TODO: test with intercept
+    # TODO: test with multiple responses
+    # basic testing
+    yield check_regressors_train
+    yield check_regressor_data_not_an_array
+    yield check_estimators_partial_fit_n_features
+    yield check_regressors_no_decision_function
+    # Test that estimators can be pickled, and once pickled
+    # give the same answer as before.
+    yield check_regressors_pickle
+    if name != 'CCA':
+        # check that the regressor handles int input
+        yield check_regressors_int
+    # Test if NotFittedError is raised
+    yield check_estimators_unfitted
+
+
+def _yield_transformer_checks(name, Transformer):
+    # All transformers should either deal with sparse data or raise an
+    # exception with type TypeError and an intelligible error message
+    yield check_transformer_pickle
+    if name not in ['AdditiveChi2Sampler', 'Binarizer', 'Normalizer',
+                    'PLSCanonical', 'PLSRegression', 'CCA', 'PLSSVD']:
+        yield check_transformer_data_not_an_array
+    # these don't actually fit the data, so don't raise errors
+    if name not in ['AdditiveChi2Sampler', 'Binarizer', 'Normalizer']:
+        # basic tests
+        yield check_transformer_general
+        yield check_transformers_unfitted
+
+
+def _yield_clustering_checks(name, Clusterer):
+    yield check_clusterer_compute_labels_predict
+    if name not in ('WardAgglomeration', "FeatureAgglomeration"):
+        # this is clustering on the features
+        # let's not test that here.
+        yield check_clustering
+        yield check_estimators_partial_fit_n_features
+
+
+def _yield_all_checks(name, Estimator):
+    #yield check_parameters_default_constructible, name, Estimator
+    for check in _yield_non_meta_checks(name, Estimator):
+        yield check
+    if issubclass(Estimator, ClassifierMixin):
```

Comment:
For transformer, I think checking for `fit_transform` or `transform` should be pretty reliable. Not sure if we need to add a new `is_transformer` helper, though.